specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2023-08-01-preview/machineLearningServices.json

```diff
@@ -3174,6 +3196,29 @@
         }
       }
     },
+    "JupyterKernelConfig": {
+      "type": "object",
+      "description": "Jupyter kernel configuration",
+      "properties": {
+        "argv": {
+          "type": "array",
+          "description": "Argument to the the runtime",
+          "items": {
+            "type": "string",
+            "description": "argument values"
+          }
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Display name of the kernel"
+        },
+        "language": {
+          "type": "string",
+          "description": "Language of the kernel"
+        }
+      },
+      "additionalProperties": true
```

Comment:
>additionalProperties [](http://example.com/codeflow?start=7&length=20)

Try to avoid this pattern wherever possible. Untyped/unschematized properties are difficult and dangerous for clients to consume. Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

Reply:
Currently we are using this properties bag to extend new features in private preview. We will consider removing it during GA

Reply:
I would suggest removing it now and monitor your preview requirements to release newer preview version with more specific properties. Using additonalProperties is not recommended for this purpose.